src/main/java/amidst/gui/profileselect/ProfileComponent.java

```diff
@@ -61,24 +62,33 @@ private void updateIfNecessary() {
 				versionNameX = width - 40 - versionFontMetrics.stringWidth(versionName);
 				oldWidth = width;
 				oldVersionName = versionName;
-				oldProfileName = createProfileName(profileName, versionNameX - 25);
+				oldProfileName = createProfileName(profileName, versionNameX - 25, 40);
 			}
 		}
 
 		@CalledOnlyBy(AmidstThread.EDT)
-		private String createProfileName(String profileName, int maxWidth) {
+		private String createProfileName(String profileName, int maxWidth, int xOffset) {
 			String result = profileName;
-			if (profileFontMetrics.stringWidth(result) > maxWidth) {
+			if (profileFontMetrics.stringWidth(result) + xOffset > maxWidth) {
 				int widthSum = 0;
 				for (int i = 0; i < result.length(); i++) {
 					widthSum += profileFontMetrics.charWidth(result.charAt(i));
-					if (widthSum > maxWidth) {
+					if (widthSum + xOffset > maxWidth) {
 						return result.substring(0, i) + "...";
 					}
 				}
 			}
 			return result;
 		}
+		
+		@CalledOnlyBy(AmidstThread.EDT)
+		private void drawProfileIcon(Graphics2D g2d) {
```

Comment:
The `scaledIcon` should probably be cached, to avoid reloading it each frame.

Reply:
Just fixed this I think.